runtime/restart/monitor/monitor.go

```diff
@@ -213,15 +215,34 @@ func (m *monitor) monitor(ctx context.Context) ([]change, error) {
 			return nil, err
 		}
 		desiredStatus := containerd.ProcessStatus(labels[restart.StatusLabel])
-		if m.isSameStatus(ctx, desiredStatus, c) {
+		task, err := c.Task(ctx, nil)
+		if err != nil && desiredStatus == containerd.Stopped {
 			continue
 		}
+		status, err := task.Status(ctx)
+		if err != nil && desiredStatus == containerd.Stopped {
+			continue
+		}
+		if desiredStatus == status.Status {
+			continue
+		}
+
 		switch desiredStatus {
 		case containerd.Running:
+			policy, err := restart.NewPolicy(labels[restart.PolicyLabel])
+			if err != nil {
+				logrus.WithError(err).Error("monitor policy")
+				continue
+			}
+			restartCount, _ := strconv.Atoi(labels[restart.CountLabel])
```

Comment:
Shouldn't we check the error here?

Reply:
IMHO, If it is an invalid restart count, we can treat it as 0. This is also for compatibility without label values.

Reply:
It would be better to handle "" specifically and log the error from Atoi().

Software that is close to users (e.g. nerdctl) could have [Do What I Mean](https://en.wikipedia.org/wiki/DWIM) behavior, but containerd is used from other software and end-users wouldn't have a direct feedback. Implicitly ignoring invalid inputs makes troubleshooting harder.